### ISSUE TITLE
yawsso needs to ignore the expiresAt parameter in .aws/sso/cache/<thing>.json

### ISSUE THREAD

**Issue by et304383** [victorskl/yawsso#90]
I have been dealing with AWS support for a while and it appears that the expiresAt parameter is not accurate as the refresh token should still be valid.

Yet yawsso just exits out whenever you try to run it after the expiresAt value has passed (which is usually only one hour).

So please just ignore the value and continue to ask for new credentials just as the auto refresh would work within the CLI.

**Comment by et304383** [comment]
The code in question that needs to be removed:

https://github.com/victorskl/yawsso/blob/main/yawsso/core.py#L97

**Comment by victorskl** [comment]
Ok, let me access the situation a bit further. Agree on the view that it might be counterproductive to check expiry while we can just refresh the token.

**Comment by victorskl** [comment]
I can reproduce this. I will make changes as advised - i.e. removing sso expiresAt check from yawsso.

**Comment by victorskl** [comment]
umm, i take it back. i have to understand a bit more how aws sso cli cache file format work.

**Comment by et304383** [comment]
> umm, i take it back. i have to understand a bit more how aws sso cli cache file format work.

It's a little confusing but basically it's a token that's good for refreshing temporary role credentials. It's good for an hour.

When it expires, the SDK refreshes the SSO token with the refresh token.

To be honest I think you should be consistent like all other tools and just try the operation. If it doesn't work downstream because the refresh token expired an appropriate error will be thrown from the CLI call. You shouldn't need to worry about that.

**Comment by victorskl** [comment]
Anyhow, let me keep tinkering around to see how to work around and, make use of refreshToken. At the last resort, I'll have to add boto as dependencies...

**Comment by victorskl** [comment]
Correction; 

It needs to use a `refreshToken` to refresh `accessToken` in sso cache file. It seems the cache value of `accessToken` from `~/.aws/sso/cache/<session_id>.json` hasn't refreshed.

I'm still investigating/debugging this... how aws cli handling `refreshToken`. It might be that it just refresh `accessToken` on-the-fly; instead of re-updating `accessToken` inside sso cache json file.

**Comment by et304383** [comment]
If it helps I have a work around right now where I delete my cli cache and then call aws sts get-caller-identity (against any profile) to force refresh of the SSO token if necessary as internally the CLI will first try to refresh the role credentials and if needed refresh the SSO token. 

Then I call yawsso

**Comment by victorskl** [comment]
Yup. Thinking very similarly around making use of `aws sts get-caller-identity`...

CLI has its own cache `ls -lrt ~/.aws/cli/cache`; it seems to use `refreshToken` to only refresh access key token there. It does not reflect back into `ls -lrt ~/.aws/sso/cache` sso session cache.

Ok, getting there. Thanks for sharing your work around. I will see to it.

**Comment by et304383** [comment]
Any updates here @victorskl ?

**Comment by victorskl** [comment]
Sorry folks. Out a bit with extended camping trip. I am back for the year. I will resume working on this.

Think, I have an idea/solution on the fix. Will make RC this weekend.

**Comment by victorskl** [comment]
It should fix with 1.2rc1. Let me know any feedback, if any.

```
pip install -U yawsso==1.2.0rc1

yawsso --version
yawsso 1.2.0rc1
```

**Comment by victorskl** [comment]
The yawsso-1.2.0 out.!

```
pip install -U yawsso==1.2.0
```

This should attempt auto-refresh as long as the refresh token lives...

Better yet, try `auto` subcommand; this will auto login as last resort when the refresh token itself has expired...
```
yawsso auto --profile dev
```
